internal/docker/docker.go

```diff
@@ -62,6 +65,51 @@ type Docker struct {
 	run func(args ...string) error
 }
 
+// GenerateRequest contains all the information required for a language
+// container to run the generate command.
+type GenerateRequest struct {
+	// repo is a pointer to the [gitrepo.Repository] struct, representing the
+	// language repository.
+	repo *gitrepo.Repository
```

Comment:
nit, the only information this package needs from this object is the dir. Maybe make this field a string and called RepoDir. That way you can decouple this whole package from needing to import gitrepo

Reply:
Done.